src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -21,27 +43,301 @@ const ActorInit En_Kakasi_InitVars = {
     (ActorFunc)EnKakasi_Update,
     (ActorFunc)EnKakasi_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi/EnKakasi_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kakasi/EnKakasi_Init.s")
+void EnKakasi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    //! @bug Skelanime_Free is not called
+}
+
+void EnKakasi_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKakasi* this = THIS;
+
+    osSyncPrintf("\n\n");
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ Ｌｅｔ’ｓ ＤＡＮＣＥ！ ☆☆☆☆☆ %f\n" VT_RST, this->actor.posRot.pos.y);
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.unk_1F = 6;
+    SkelAnime_InitSV(globalCtx, &this->skelanime, &D_060065B0, &D_06000214, NULL, NULL, 0);
+
+    this->rot = this->actor.posRot.rot;
+    this->actor.flags |= 0x400;
+    this->actor.colChkInfo.mass = 0xFF;
+
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = func_80A8F660;
+}
+
+void func_80A8F28C(EnKakasi* this) {
+    this->unk_1A4 = 0;
+    this->skelanime.animPlaybackSpeed = 0.0f;
+    this->unk_1A8 = this->unk_1AC = 0x0;
+
+    Math_SmoothDownscaleMaxF(&this->skelanime.animCurrentFrame, 0.5f, 1.0f);
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.x, this->rot.x, 0x5, 0x2710, 0x0);
```

Comment:
I can understand why we should probably use hex for rotations (tho I don't really agree with it), but I don't think we should use it for things like 0x0 or 0x5.

Reply:
Mostly saying this now so we don't start going in the wrong direction. I know you changed it because of a previous review comment, so sorry about that.

Reply:
I also find weird that rotations tbh, so small values decimal, big values hex? what's a big value, though?

Reply:
I'd say 10 or above, since that's where hex starts making a difference other than the leading 0x

Reply:
👍

Reply:
yeah usually any hex value < 10 is written without the 0x prefix, not just rotations

rotations in hex make sense imo because the usual values like 90° are nice in hex (`0x4000`)
So you understand easier what `0x2710` (here) is, somewhere between 45° (`0x2000`) and ~70° (`0x3000`)
where `10000 = 0x2710` is to be compared to `8192 = 0x2000` and `12288 = 0x3000` which are way less nice values in decimal

it's just like radians, I have no clue what `0.1` radians is, but you know `pi / 4` radians is 45°